docs/streams/upgrade-guide.html

```diff
@@ -76,7 +76,14 @@ <h3><a id="streams_api_changes_240" href="#streams_api_changes_240">Streams API
 
     <!-- Placeholder KIP-213 -->
     <!-- Placeholder KIP-307 -->
-    <!-- Placeholder KIP-479 -->
+    <p>
+        Before the 2.4.0 version of Kafa Streams, users of the DSL could not name the state stores involved in a stream-stream join.  If users added DSL operations before of the 
+        join, the internal names of the state stores would shift, requiring an application reset when redeploying.  In the 2.4.0 release, Kafka Streams adds the <code>StreamJoined</code>
+        object, which gives users the ability to name the join processor, repartition topic(s) (if a repartition is required), and the state stores involved in the join.  Also, by naming the state stores, the changelog topics backing the state stores are named as well.
+        <br/>
+        Another feature delivered by <code>StreamJoined</code> is that you can now configure the type of state store used in the join.  You can now elect to use in-memory stores or custom state stores for a stream-stream join.  Note that the provided stores will not be available for querying via Interactive Queries.  With the addition of <code>StreamJoined</code>, stream-stream join operations using <code>Joined</code> have been deprecated. Please switch over to stream-stream join methods using the new overloaded methods.  You can get more details from 
```

Comment:
`You can now elect ` -> `You can elect` (the `now` is repetitive as it's used in the sentence before already)

Also, this line is too long

Reply:
`Note that the provided stores will not be available for querying via Interactive Queries` -> this is also true if the store in only named (might be worth to mention in the paragraph above already)?